test/correctness/simd_op_check.cpp

```diff
@@ -1871,6 +1871,16 @@ struct Test {
         check("vmpa(v*.h,r*.b)", hvx_width/2, 2*i32(i16_1) + 3*i32(i16_2));
         check("v*.w += vmpa(v*.h,r*.b)", hvx_width/2, 2*i32(i16_1) + 3*i32(i16_2) + i32_1);
 
+        check("v*:*.h = vtmpy(v*:*.ub, r*.b)", hvx_width/1, 2*i16(in_u8(x)) + 3*i16(in_u8(x+1)) + i16(in_u8(x+2)));
```

Comment:
Good to hear. In that case, could you write the simd_op_check that way? That will give coverage to the common case that occurs in real code.